hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -1721,7 +1722,7 @@ public String generateSelectVersionString() {
 		SimpleSelect select = new SimpleSelect( getFactory().getDialect() )
 				.setTableName( getVersionedTableName() );
 		if ( isVersioned() ) {
-			select.addColumn( versionColumnName );
+			select.addColumn( getVersionColumnName(), VERSION_COLUMN_ALIAS );
 		}
 		else {
```

Comment:
I suspect that there will be a similar failure when an entity is not versioned, but it has an ID column that is quoted. I did look into that, but apparently that use case is broken in a different way (NPE is thrown by `EntityVerifyVersionProcess#doBeforeTransactionCompletion`).

It's OK with me if this other use case is not fixed here.

Reply:
That code assumes that the entity is versioned.  This comes back to my comment below that I'm not sure this should be done at all if the entity is not versioned

Reply:
@sebersole, I guess it wasn't clear, but I was referring to the same block you mention. I already tried what you suggested, but it fails as I described above.

<  I did look into that, but apparently that use case is broken in a different way (NPE is thrown by EntityVerifyVersionProcess#doBeforeTransactionCompletion).